### PR TITLE
qemu_vm:windows 32bit guest's maximum mem limit to 3G

### PR DESCRIPTION
**Pull request by aliang123**
qemu_vm:
1.modify function add_memorys for 32bit windows guest
2.correct pep8 error

Signed-off-by: Aihua Liang<aliang@redhat.com>

ID:1388321,1388309

### CODE REVIEW

**Comment by aliang123** [comment]
ID:1388321

**Comment by ngu-niny** [comment]
LGTM.
Acked-by: Nini Gu <ngu@redhat.com>

**Comment by xutian** [comment]
Nacked, this should be limited in  windows-32.cfg

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by XueqiangWei <xuwei@redhat.com>

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -892,21 +892,25 @@ def add_memorys(devices, params):
...
                 return devices
 
             cmdline = "-m %s" % ",".join(options)
             dev = StrDev("mem", cmdline=cmdline)
             devices.insert(dev)
-            for name in params.objects("mem_devs"):
-                mem_params = params.object_params(name)
-                memdevs = devices.memory_define_by_params(mem_params, name)
+            for name in mem_params.objects("mem_devs"):
+                mem_dev_params = params.object_params(name)
```
Comment (xutian) [inline virttest/qemu_vm.py]: drop this line, because you var looks meanness

Comment (xutian) [inline virttest/qemu_vm.py]: devices.memory_define_by_params(mem_params.object_params(name), name)

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -892,21 +892,25 @@ def add_memorys(devices, params):
...
+            mem = float(mem_params.get("mem", 0))
+            maxmem = float(utils_misc.normalize_data_size(mem_params.get("maxmem", 0), order_magnitude="M"))
+            actual_mem = int(min(mem, maxmem))
+            params["mem"] = actual_mem
+            if mem_params.get("mem"):
                 options.append("%s" % params["mem"])
-            if params.get("slots") and params.get("maxmem"):
-                options.append("slots=%s" % params["slots"])
-                options.append("maxmem=%s" % params["maxmem"])
+            if mem_params.get("slots") and mem_params.get("maxmem"):
```
Comment (PyLearner) [inline virttest/qemu_vm.py]: why in line 897 ,you get it like mem_params.get("maxmem", 0) and in this line you invoke it mem_params.get("maxmem") , could you tell what is the difference ,thanks.

**Comment by ngu-niny** [comment]
@aliang123 Failed to start vm in the test:

05:19:00 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/avocado_vt/test.py:420
05:19:00 ERROR| Traceback (most recent call last):
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
05:19:00 ERROR|     self._runTest()
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/avocado_vt/test.py", line 327, in _runTest
05:19:00 ERROR|     params = env_process.preprocess(self, params, env)
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
05:19:00 ERROR|     return fn(*args, **kwargs)
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/env_process.py", line 851, in preprocess
05:19:00 ERROR|     process(test, params, env, preprocess_image, preprocess_vm)
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/env_process.py", line 546, in process
05:19:00 ERROR|     _call_vm_func()
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/env_process.py", line 515, in _call_vm_func
05:19:00 ERROR|     vm_func(test, vm_params, env, vm_name)
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/env_process.py", line 183, in preprocess_vm
05:19:00 ERROR|     migration_exec_cmd=params.get("migration_exec_cmd_dst"))
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
05:19:00 ERROR|     return fn(*args, **kwargs)
05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/virttest/qemu_vm.py", line 2654, in create
05:19:00 ERROR|     raise virt_vm.VMStartError(self.name, 'Error occurred while '
05:19:00 ERROR| VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.
05:19:00 ERROR|
05:19:00 ERROR| Traceback (most recent call last):

05:19:00 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-43.0-py2.7.egg/avocado_vt/test.py", line 434, in _run_avocado
    raise test_exception

05:19:00 ERROR| VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.

05:19:00 ERROR| ERROR 1-smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.boot -> VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.

**Comment by aliang123** [comment]
Test cmd:
   python ConfigTest.py --guestname=Win7 --platform=i386 --verbose=yes --testcase=balloon_service
Test Result:
  PASS 1-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win7.i386.sp1.io-github-autotest-qemu.balloon_service
 ********************Details:****************
params:
06:52:20 DEBUG|     maxmem = 3G 
06:52:20 DEBUG|     mem = 8192
06:52:20 DEBUG|     mem_chk_cmd = wmic MemoryChip get Capacity
06:52:20 DEBUG|     mem_chk_cur_cmd = wmic memphysical

qemu cmd:
    -device virtio-blk-pci,id=image1,drive=drive_image1,bootindex=0,bus=pci.0,addr=03 \
    -m 3072  \
    -smp 4,maxcpus=4,cores=2,threads=1,sockets=2  \

**Comment by ngu-niny** [comment]
@aliang123 The make_create_command() issue still exists on RHEL.7.3 and Win2012.r2.x86_64 guests.

**Comment by aliang123** [comment]
@ngu-niny , ok, i'll check it again.

**Comment by aliang123** [comment]
Win8.1-x86_64:
   Test cmd:
       python ConfigTest.py --guestname=Win8..1 --platform=x86_64 --verbose=yes --testcase=balloon_service 
   Test Result:
      # python scan_results.py 
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_service' Finshed 2016-12-08 10:33:27 2016-12-08 10:39:52 PASS 384.663256

Win2012R2-x86_64:
  Test cmd:
     python ConfigTest.py --guestname=Win2012..r2 --platform=x86_64 --verbose=yes --testcase=balloon_service
  Test Result:
      # python scan_results.py -d /root/avocado/job-results/job-2016-12-07T20.14-26feec1
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_service' Finshed 2016-12-07 20:14:12 2016-12-07 20:20:30 PASS 377.863624

**Comment by aliang123** [comment]
RHEL.7.3-X86_64:
  Test cmd:
     python ConfigTest.py --guestname=RHEL.7.3 --platform=x86_64 --verbose=yes --testcase=balloon_check.balloon_evict.balloon-reboot
  Test Result:
    '3-smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.balloon_check.balloon_evict.balloon-reboot' Finshed 2016-12-07 22:24:14 2016-12-07 22:29:47 PASS 332.89857

Win7.x86
  Test cmd:
    python ConfigTest.py --guestname=Win7 --platform=i386 --verbose=yes --testcase=balloon_service
  Test Result:
    # python scan_results.py -d /root/avocado/job-results/job-2016-12-07T21.08-2e97d9e
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
'1-smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win7.i386.sp1.io-github-autotest-qemu.balloon_service' Finshed 2016-12-07 21:08:24 2016-12-07 21:21:30 PASS 786.641484

**Comment by aliang123** [comment]
@ngu-niny, please have to check again,thanks

**Comment by ngu-niny** [comment]
3G used now for 32bit Windows client guest:
    -drive id=drive_image1,if=none,snapshot=off,aio=native,cache=none,format=qcow2,file=/usr/share/avocado/data/avocado-vt/images/win8-32.1-virtio-scsi.qcow2 \
    -device scsi-hd,id=image1,drive=drive_image1 \
    -device virtio-net-pci,mac=9a:05:06:07:08:09,id=iddPLFCY,vectors=4,netdev=idFuD1Gc,bus=pci.0,addr=04  \
    -netdev tap,id=idFuD1Gc,vhost=on,vhostfd=20,fd=19 \
    -m 3072 

And test pass:
PASS 1-smp_16.32768m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.balloon_check.balloon_enlarge.balloon-reboot
PASS 2-smp_16.32768m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win8.i386.1.io-github-autotest-qemu.balloon_check.balloon_enlarge.balloon-reboot
PASS 3-smp_16.32768m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_check.balloon_enlarge.balloon-reboot
PASS 4-smp_16.32768m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_check.balloon_enlarge.balloon-reboot

Ack-by: Nini Gu <ngu@redhat.com>

**Comment by xutian** [comment]
Looks good to me.

**Pull request merged**